cupy/padding/pad.py

```diff
@@ -2,15 +2,37 @@
 
 
 def pad(array, pad_width, mode=None, **kwargs):
-    if not isinstance(pad_width, int):
-        raise NotImplementedError
-
+    if not isinstance(array, (list, tuple, cupy.ndarray)):
+        return cupy.array(array)
+    if isinstance(array, (list, tuple)):
+        array = cupy.array(array)
     if mode != 'constant':
         raise NotImplementedError
-
-    shape = tuple(s + pad_width * 2 for s in array.shape)
-    # TODO(unno): Fix fill value
-    ret = cupy.full(shape, 0, dtype=array.dtype)
-    index = tuple(slice(pad_width, pad_width + s) for s in array.shape)
+    constant_value = kwargs.pop('constant_values', None)
+    if len(kwargs) != 0:
+        raise NotImplementedError
+    if constant_value is None:
+        constant_value = 0
+    if not isinstance(constant_value, int):
+        raise NotImplementedError
+    if not cupy.asarray(pad_width).dtype.kind == 'i':
+        raise TypeError('`pad_width` must be of integral type.')
+    if cupy.asarray(pad_width).ndim > 1:
```

Comment:
You can move this condition to inside the `else` below as follows so that we do not have to check `ndim` when the type of `pad_width` is int.
```python
if isinstance(pad_width, int):
    ....
else:
    if cupy.asarray(pad_width).ndim > 1:
        ...
    if len(pad_width) == 1:
        ...
    elif len(pad_width) == 2:
        ...
    else:
        ...
```